scripts/machine-launch-script.sh

```diff
@@ -1,85 +1,265 @@
+    fi
+
+    # https://conda-forge.org/feedstock-outputs/ 
+    #	filterable list of all conda-forge packages
+    # https://conda-forge.org/#contribute 
+    #	instructions on adding a recipe
+    # https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications
+    #   documentation on package_spec syntax
+    CONDA_PACKAGE_SPECS=()
+
+
+    # https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#using-the-compiler-packages
+    #    for notes on using the conda compilers
+    CONDA_PACKAGE_SPECS=( gcc_linux-64 gxx_linux-64 )
+
+    # if we want to install prebuilt riscv toolchain, the bdutrosv channel has riscv-gnu-toolchain
+
+    # poky deps
+    CONDA_PACKAGE_SPECS+=( python=3.8 patch texinfo subversion chrpath git wget )
+    # qemu deps
+    CONDA_PACKAGE_SPECS+=( gtk3 glib pkg-config bison flex )
+    # qemu also requires being built against kernel-headers >= 2.6.38 because of it's use of
+    # MADV_NOHUGEPAGE in a call to madvise.  See:
+    # https://man7.org/linux/man-pages/man2/madvise.2.html
+    # https://conda-forge.org/docs/maintainer/knowledge_base.html#using-centos-7
+    # obvi this would need to be made linux-specific if we supported other MacOS or Windows
+    CONDA_PACKAGE_SPECS+=( "kernel-headers_linux-64>=2.6.38" )
+    # firemarshal deps
+    CONDA_PACKAGE_SPECS+=( rsync )
+    # cross-compile glibc 2.28+ deps
+    CONDA_PACKAGE_SPECS+=( make )
+    # other misc deps
+    # conda-forge splits out the shared library into libzlib
+    # not sure if we need both, but I'm including them for now
+    CONDA_PACKAGE_SPECS+=(
+        sbt \
+        ca-certificates \
+        mosh \
+        gmp \
+        mpfr \
+        mpc \
+        zlib \
+	libzlib \
+        vim \
+        git  \
+        openjdk \
+        gengetopt \
+        libffi \
+        expat \
+        libusb1 \
+        ncurses \
+        cmake \
+        graphviz \
+        expect
+        dtc \
+	verilator==4.034 \
+    )
+
+    # python packages
+    # While it is possible to install using pip after creating the
+    # conda environment, pip's dependency resolution can conflict with
+    # conda and create broken environments.  It's best to use the conda
+    # packages so that the environment is consistent
+    CONDA_PACKAGE_SPECS+=( \
+	boto3==1.20.21 \
+        colorama==0.4.3 \
+	argcomplete==1.12.3 \
+	python-graphviz==0.19 \
+	pyparsing==3.0.6 \
+	numpy==1.19.5 \
+	kiwisolver==1.3.1 \
+	matplotlib==3.3.4 \
+	pandas==1.1.5 \
+	awscli==1.22.21 \
+	pytest==6.2.5 \
+	pytest-dependency==0.5.1 \
+	pytest-mock==3.7.0 \
+	moto==2.2.19 \
+	pyyaml==5.4.1 \
+	mypy==0.931 \
+	types-pyyaml==6.0.4 \
+	boto3-stubs==1.21.6 \
+	botocore-stubs==1.24.7 \
+	mypy-boto3-s3==1.21.0 \
+    )
+
+    if [[ "$CONDA_ENV_NAME" == "base" ]]; then
+	# we default to installing into the base environment against best practice because it
```

Comment:
I know you've explained to me orally, but can you explain how it simplifies it here.

Reply:
`conda init ` either adds `/etc/profile.d/conda.sh` (the default behavior of this script when installing to `/opt/conda` as root) or to the users `~/.bashrc`.   I should note that conda also has support for other shells but I didn't introduce that in this commit.  baby steps.

By default, the code added puts `/opt/conda/condabin` into your path, makes a `conda` shell function in your shell (`conda activate` has to be a shell function to be able to manipulate the current shell environment) and it activates the `base` environment.

I was trying to do a 'drop-in' replacement of using the system packages where any user who logged into the server would have `/opt/conda/bin` in their path and we wouldn't have to ripple knowledge of `conda` further into the flow (e.g. like having `env.sh` do something conda-specific).

We may decide that we want more of the flow to need to know about conda but if you want to decouple the rest of the flow from conda, this is what I came up with as a first cut.  The only cost was installing more stuff into the `base` environment which is frowned upon because you can break conda itself when you do certain things to the `base` environment and then fixing it requires re-installing the base environment from the installer tar-script.

I actually started out installing everything into a `machine-launch` environment and then wanting to add `conda activate machine-launch` to env.sh and that leads to wanting to add another option to `build-setup.sh` where you can specify which environment you want to use.

Reply:
Sorry, i wasn't clear, by explain how it simplifies it here, i meant as a comment in the code..

Reply:
You could just as well link to your GH comment in your code comment though lol

Reply:
Sure.   I can do that and it will also be here in the PR! :)

Reply:
```suggestion
	# see also: https://github.com/firesim/firesim/pull/986#discussion_r830207675
	# we default to installing into the base environment against best practice because it
```